### PR TITLE
Update BadgeField to respect dynamic text sizing

### PR DESCRIPTION
**Pull request by mischreiber**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

- Removed fixed-height assumptions in `BadgeField`, allowing the UI to respond to dynamic text sizing.
- Replaced instances of `UILabel` with `FluentUI.Label`, resulting in automatic scaling.

### Verification

Verified the following:
- Small, large, and extra-extra large sizing
- UI reflowed appropriately when resized while visible
- Adding new badges via textfield input
- Dark mode

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| before | after |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] VoiceOver and Keyboard Accessibility
- [x] Internationalization and Right to Left layouts
- [x] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/670)

### CODE REVIEW

File: ios/FluentUI/Badge Field/BadgeField.swift [hunk 1]
```diff
@@ -786,6 +795,11 @@ open class BadgeField: UIView {
         originalDeviceOrientation = UIDevice.current.orientation
     }
 
+    @objc private func handleContentSizeCategoryDidChange() {
+        invalidateIntrinsicContentSize()
```
Comment (harrieshin) [inline ios/FluentUI/Badge Field/BadgeField.swift:799]: should the order swap so you calculate the height first before triggering invalidation?

**Comment by harrieshin** [comment]
while this is a fine change and we should merge it, is there a limitation of this control to actually use auto layout rather than frame calculation?

**Review by harrieshin — APPROVED** [review]

**Comment by mischreiber** [comment]
> while this is a fine change and we should merge it, is there a limitation of this control to actually use auto layout rather than frame calculation?

I think the problem is that the badges flow from line to line, which is really hard to do with auto layout. If I were to rewrite this today, I'd explore using a `UICollectionViewFlowLayout`(though it's possible that the original author did, and found some limitation).

File: ios/FluentUI/Badge Field/BadgeField.swift (hunk 1 above)
Comment (mischreiber) [reply ios/FluentUI/Badge Field/BadgeField.swift:799]: Shouldn't matter. `invalidateIntrinsicContentSize` sets a flag on the view that will require a re-layout on the next layout pass (effectively the next iteration of the run loop), but doesn't directly trigger a synchronous layout. And all this is done on the main thread, so there's no race condition to worry about.

**Review by rdeassis — APPROVED** [review]

**Comment by rdeassis** [comment]
Shouldn't the badges also have unlimited number of lines in the scenario where the badge field has unlimited number of lines?

File: ios/FluentUI/Badge Field/BadgeField.swift [hunk 2]
```diff
@@ -436,18 +438,20 @@ open class BadgeField: UIView {
         }
     }
 
-    @objc open func heightThatFits(numberOfLines: Int) -> CGFloat {
+    @objc open func heightThatFits(badgeHeight: CGFloat, numberOfLines: Int) -> CGFloat {
         // Vertical margin not applicable to top line
-        let heightForAllLines = CGFloat(numberOfLines) * (Constants.badgeHeight + Constants.badgeMarginVertical)
+        let heightForAllLines = CGFloat(numberOfLines) * (badgeHeight + Constants.badgeMarginVertical)
```
Comment (rdeassis) [inline ios/FluentUI/Badge Field/BadgeField.swift:443]: Shouldn't the badges also have unlimited number of lines in the scenario where the badge field has unlimited number of lines?

Comment (mischreiber) [reply ios/FluentUI/Badge Field/BadgeField.swift:443]: Not sure. I've never seen a `BadgeView` with multiple line support, and even if they did, there's still a difference between "let `BadgeView`s flow across multiple lines" and "each `BadgeView` can use as many lines as it wants".

I'm inclined to leave the behavior as-is, but that might be an interesting follow-up for design in the future.

**Pull request merged**